### PR TITLE
feat: remove filecoin deps

### PR DESCRIPTION
**Pull request by gallexis**
closes: #171

### CODE REVIEW

**Comment by gallexis** [comment]
@tchardin @alexander-camuto This is a working branch where I cleansed everything I could.
There is still some stuff to be removed but I let you handle that since I'm not sure about how to do it.
For some reason we still need the ./extern/filecoin-ffi folder, on my previous branch I deleted it but i could not compile :
```
# pkg-config --cflags  -- /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc /Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc
Failed to open '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc': No such file or directory
No package '/Users/alexisgallepe/go/pkg/mod/github.com/filecoin-project/filecoin-ffi@v0.30.4-0.20200910194244-f640612a1a1f/generated/../filcrypto.pc' found
```
Looks like it needs a filcrypto.pc file

**Comment by gallexis** [comment]
So we have some filecoin related libs such as github.com/filecoin-project/go-commp-utils that need filecoin-ffi

**Comment by tchardin** [comment]
@gallexis can you keep all the files in `/build` untouched. Also we shouldn't need `go-commp-utils` so you can remove that too.

**Comment by gallexis** [comment]
@tchardin actually we have all these go modules in the repo ([go.mod](https://github.com/myelnet/pop/blob/main/go.mod)) that require filecoin-ffi and they are all currently used in our codebase : 
```go
	github.com/filecoin-project/go-address v0.0.5
	github.com/filecoin-project/go-amt-ipld/v2 v2.1.1-0.20201006184820-924ee87a1349 // indirect
	github.com/filecoin-project/go-cbor-util v0.0.0-20201016124514-d0bbec7bfcc4
	github.com/filecoin-project/go-commp-utils v0.1.1-0.20210427191551-70bf140d31c7
	github.com/filecoin-project/go-crypto v0.0.0-20191218222705-effae4ea9f03
	github.com/filecoin-project/go-data-transfer v1.7.0
	github.com/filecoin-project/go-fil-markets v1.6.0-rc1
	github.com/filecoin-project/go-hamt-ipld/v3 v3.0.1
	github.com/filecoin-project/go-jsonrpc v0.1.2
	github.com/filecoin-project/go-multistore v0.0.3
	github.com/filecoin-project/go-state-types v0.1.1-0.20210506134452-99b279731c48
	github.com/filecoin-project/go-statemachine v0.0.0-20200925024713-05bd7c71fbfe
	github.com/filecoin-project/specs-actors/v5 v5.0.0-20210512015452-4fe3889fff57
```

**Comment by tchardin** [comment]
@gallexis that's not exactly accurate, the only modules that do depend on ffi are `go-commp-utils` and `go-fil-markets` and we should be getting rid of them. You can remove the storage module and whatever is requesting it. I don't think we have any hard dependencies but let me know if we do.

**Comment by gallexis** [comment]
@tchardin Alright I'll try to remove all the code related to github.com/filecoin-project/*

File: retrieval/client/fsm_test.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ type mockClientEnvironment struct {
 }
 
 func (e *mockClientEnvironment) OpenDataTransfer(ctx context.Context, to peer.ID, proposal *deal.Proposal) (datatransfer.ChannelID, error) {
-	return datatransfer.ChannelID{ID: datatransfer.TransferID(rand.Uint64()), Responder: to, Initiator: testnet.GeneratePeers(1)[0]}, e.OpenDataTransferError
+	return datatransfer.ChannelID{ID: datatransfer.TransferID(rand.Uint64()), Responder: to}, e.OpenDataTransferError
 }
```
Comment (gallexis) [inline retrieval/client/fsm_test.go:46]: @tchardin not sure about this one

Comment (tchardin) [reply retrieval/client/fsm_test.go:46]: @gallexis just make a util function to generate a random peer? You just need a dummy peer ID

File: filecoin/testutils.go [hunk 2]
```diff
@@ -34,12 +34,12 @@ func testBlockHeader() *BlockHeader {
 		},
 		Parents:               []cid.Cid{c, c},
 		ParentMessageReceipts: c,
-		BLSAggregate:          &crypto.Signature{Type: crypto.SigTypeBLS, Data: []byte("bls signature")},
+		BLSAggregate:          &crypto.Signature{Type: crypto.SigTypeSecp256k1, Data: []byte("secp256k1 signature")},
```
Comment (tchardin) [inline filecoin/testutils.go:37]: For accuracy's sake can you leave this as BLS? it doesn't matter in this test but this should be a BLS signature and same for below.

File: filecoin/types.go [hunk 3]
```diff
@@ -439,10 +439,6 @@ type SignedMessage struct {
 }
 
 func (sm *SignedMessage) ToStorageBlock() (block.Block, error) {
-	if sm.Signature.Type == crypto.SigTypeBLS {
```
Comment (tchardin) [inline filecoin/types.go:442]: This should also be left as is.

File: node/node_test.go [hunk 4]
```diff
@@ -557,19 +557,35 @@ func TestMultipleGet(t *testing.T) {
...
+	require.NoError(t, err)
+
+	f, err := os.Open(keyPath)
+	require.NoError(t, err)
+
+	data, err := io.ReadAll(f)
+	require.NoError(t, err)
+
+	// Assert that imported & exported keys are the same
+	require.EqualValues(t, h, string(data))
```
Comment (tchardin) [inline node/node_test.go:588]: Cool

File: retrieval/client/fsm_test.go [hunk 5]
```diff
@@ -42,7 +42,7 @@ type mockClientEnvironment struct {
...
 
 func (e *mockClientEnvironment) OpenDataTransfer(ctx context.Context, to peer.ID, proposal *deal.Proposal) (datatransfer.ChannelID, error) {
-	return datatransfer.ChannelID{ID: datatransfer.TransferID(rand.Uint64()), Responder: to, Initiator: testnet.GeneratePeers(1)[0]}, e.OpenDataTransferError
+	return datatransfer.ChannelID{ID: datatransfer.TransferID(rand.Uint64()), Responder: to, Initiator: GeneratePeers(1)[0]}, e.OpenDataTransferError
```
Comment (tchardin) [inline retrieval/client/fsm_test.go:45]: If GeneratePeers is only used here I would literally put: `peer.ID(1)` there and not make the method.

File: retrieval/deal/types_cbor_gen.go [hunk 6]
```diff
@@ -8,9 +8,8 @@ import (
 	"sort"
 
 	address "github.com/filecoin-project/go-address"
-	piecestore "github.com/filecoin-project/go-fil-markets/piecestore"
 	multistore "github.com/filecoin-project/go-multistore"
-	paych "github.com/filecoin-project/specs-actors/v5/actors/builtin/paych"
+	paych "github.com/filecoin-project/specs-actors/actors/builtin/paych"
```
Comment (tchardin) [inline retrieval/deal/types_cbor_gen.go:12]: Please make sure you keep `/v5` here

**Review by tchardin — COMMENTED** [review]
A few last adjustments and if all tests are passing and it's building alright with the new makefile you should be good to go!

**Review by tchardin — APPROVED** [review]
Cool just rebase and you should be good to merge! Thanks

**Comment by gallexis** [comment]
@tchardin let's merge this? 😬😁

**Comment by tchardin** [comment]
Go for it

**Pull request merged**